packages/strapi-plugin-content-manager/admin/src/components/DynamicZone/index.js

```diff
@@ -201,32 +246,40 @@ const DynamicZone = ({
               values={{ componentName: name }}
             />
           </div>
-          <ComponentsPicker isOpen={isOpen}>
-            <div>
-              <p className="componentPickerTitle">
-                <FormattedMessage id={`${pluginId}.components.DynamicZone.pick-compo`} />
-              </p>
-              <div className="componentsList">
-                {dynamicZoneAvailableComponents.map(componentUid => {
-                  const { icon, name: friendlyName } = getDynamicComponentInfos(componentUid);
-
-                  return (
-                    <DynamicComponentCard
-                      key={componentUid}
-                      componentUid={componentUid}
-                      friendlyName={friendlyName}
-                      icon={icon}
-                      onClick={() => {
-                        setIsOpen(false);
-                        const shouldCheckErrors = hasError;
-                        addComponentToDynamicZone(name, componentUid, shouldCheckErrors);
-                      }}
-                    />
-                  );
-                })}
+          <Collapse isOpen={isOpen}>
+            <ComponentsPicker>
+              <div>
+                <p className="componentPickerTitle">
+                  <FormattedMessage id={`${pluginId}.components.DynamicZone.pick-compo`} />
+                </p>
+                <div className="categoriesList">
+                  {dynamicComponentCategories.map(({ category, components }, index) => {
```

Comment:
Therefore, this needs to be updated as follow:

```
{Object.keys(dynamicComponentCategories).map((categoryName, index) => {
  const components = dynamicComponentCategories[categoryName];

  return (
    <CategoryItem
      key={categoryName}
      category={categoryName}
      components={components}
      isOpen={categoryToOpen === categoryName}
      isFirst={index === 0}
      onClickToggle={() => {
       // Detailed below
        handleClickToggle(categoryName);
      }}
      onClickComponent={componentUid => {
        setCategoryToOpen('');
        const shouldCheckErrors = hasError;
        addComponentToDynamicZone(name, componentUid, shouldCheckErrors);
      }}
    />
  );
})}
```


```
const handleClickToggle = useCallback(
  categoryName => {
    const nextCategoryToOpen = categoryToOpen === categoryName ? '' : categoryName;

    setCategoryToOpen(nextCategoryToOpen);
  },
  [categoryToOpen]
);
```

Reply:
@soupette done in 15570e056d4c6c19e4615bd214dd63fbff7df881